coalib/parsing/Globbing.py

```diff
@@ -358,49 +368,51 @@ def _iglob(pattern):
         relative_glob_function = relative_flat_glob
 
     if not dirname:
-        for file in relative_glob_function(dirname, basename):
+        for file in relative_glob_function(dirname, basename, ignored_globs):
             yield file
         return
 
     # Prevent an infinite recursion if a drive or UNC path contains
     # wildcard characters (i.e. r'\\?\C:').
     if dirname != pattern and has_wildcard(dirname):
-        dirs = iglob(dirname)
+        dirs = iglob(dirname, ignored_globs)
     else:
         dirs = [dirname]
 
     for dirname in dirs:
-        for name in relative_glob_function(dirname, basename):
+        for name in relative_glob_function(dirname, basename, ignored_globs):
             yield os.path.join(dirname, name)
 
 
 @yield_once
-def iglob(pattern):
+def iglob(pattern, ignored_globs=None):
     """
     Iterates all filesystem paths that get matched by the glob pattern.
     Syntax is equal to that of fnmatch.
 
-    :param pattern: Glob pattern with wildcards
-    :return:        Iterator that yields all file names that match pattern
+    :param pattern:       Glob pattern with wildcards
+    :param ignored_globs: List of globs to ignore when matching files
+    :return:              Iterator that yields all file names that match pattern
     """
     for pat in _iter_alternatives(pattern):
         pat = os.path.expanduser(pat)
         pat = os.path.normcase(pat)
 
         if pat.endswith(os.sep):
-            for name in _iglob(pat):
+            for name in _iglob(pat, ignored_globs):
```

Comment:
Ignored globs were filtered after all files were collected. Initially we were looking for all files and directories in the project and then remove the files and directories that were matching ignore globs.